### ISSUE TITLE
System.BadImageFormatException: 'Invalid field token 0x040000a1'

### ISSUE THREAD

**Issue by dg2k** [ewerspej/epj.ProgressBar.Maui#2]
I tried to use this with net8.0 but crashes with the above exception error. Changed the projects to net8.0 and all compile okay. Looks like a very elusive error message and was wondering if there is any workaround.

I tried a likely workaround suggested [here ](https://github.com/mono/SkiaSharp/issues/2444) but that didn't help.

- VS Version: 17.7.0 Preview 2.0
- .NET SDK Version:  8.0.100-preview.4.23260.5
- OS Version:  10.0.25393
- Tested on Android API Versions:  33, 31, 30 with identical results (all emulators)
- Tested on Physical Device Samsung M32 (API 33) -- same result.

**Comment by ewerspej** [comment]
@dg2k Thanks for reporting this. I haven't tried the control with .NET 8.0, yet. Please note that .NET 8.0 is currently in preview and will only be released in November 2023. This seems to be an issue with SkiaSharp, so I cannot do anything about it at this moment, unfortunately, because the control needs to remain backward-compatible. Once a fix is available, I may create a new preview version.

**Comment by ewerspej** [comment]
> Thank you @ewerspej for the reply. I just upgraded to net8.0 SDK version 8.0.100-preview.5.23303.2 and the problem seems to go away. So far so good, and perhaps the issue was fixed in 8.0.100-preview.5.23303.2.

Okay, great.

> An enhancement request: I like the ProgressBar control. Can the Animation code which you put in the view page be part of the control code that can be enabled or disabled via a property, such as IsAnimationEnabled? I use progress control in various pages which requires me putting the animation code in each view page. Just a suggestion and thank you for you kind help.

Technically, that is definitely possible. Please submit a separate issue for this. I need to think about the side effects of this, I want all users to be able to have their own types of animations, but providing a default one may be a good idea. You could also wrap the control inside a `ContentView`, expose your own `IsAnimationEnabled` [BindableProperty](https://learn.microsoft.com/dotnet/maui/fundamentals/bindable-properties) and then create your own animation only once.

**Comment by YMichurin** [comment]
I've installed the latest net8.0 SDK version 8.0.100-preview.5.23303.2, but got the same `System.BadImageFormatException: 'Invalid field token 0x040000a1'` error

**Comment by dg2k** [comment]
@YMichurin, that's very strange and even worrying because "it worked" doesn't really mean it worked for all.  It means there may be other factors at play then.

Now, upgrading to 8.0.100-preview.5.23303.2 showed up something for me -- Google OAuth doesn't work anymore. I was compelled to use net8.0 because there were some layout issues in net7.0 that got fixed in net8.0 but the upgrade with net8.0 always comes up with something else, constantly moving my 🧀.

I think I am now setting for net7.0 until November when net8.0 leaves the lab because most of the layout issues I had with net7.0 seem to have been fixed with latest updates. The latest **7.0.400-preview.23274.1** looks like to have addressed a number of issues that forced me to consider net8.0 in the first place.

**Comment by ewerspej** [comment]
Depending on what issues you're dealing with, you may want to check out this complementary NuGet package which contains workarounds and temporary bug fixes for several MAUI issues: https://www.nuget.org/packages/PureWeen.Maui.FixesAndWorkarounds

It's by Shane Neuville, one of the MAUI developers.

**Comment by ewerspej** [comment]
Issue is directly related to MAUI and .NET.
